Fix for issue #6042. Fix error message when phpinfo() is called.

pcmitsis opened the pull request:
Fix error message when phpinfo() is called on server that disables this function

I added a function called phpinfo_enabled at sysinfo.php (administrator/components/com_admin/models/sysinfo.php) which check if phpinfo is enabled on server. If it is enabled then getPHPInfo will execute normaly otherwise an empty string will return with no warning.

pcmitsis committed:
Fix for issue #6042. Fix error message when phpinfo() is called on server that disables this function

I added a function called phpinfo_enabled at sysinfo.php (administrator/components/com_admin/models) which check if phpinfo is enabled
on server. If it is enabled then getPHPInfo will execute normaly otherwise an empty string will return with no warning.

pcmitsis committed:
Fix whitespace for PR 6148

zero-24 commented on administrator/components/com_admin/models/sysinfo.php:
```diff
@@ -155,6 +155,19 @@ public function &getInfo()
+	 *
+	 * @return  boolean True if enabled
+	 *
+	 * @since  3.5
```
@pcmitsis i think you can add 3.4.1 here as i hope this can go into the 3.4 stage and don't need to wait for 3.5  but maybe don't get into 3.4.0 :smile:

zero-24 commented on administrator/components/com_admin/models/sysinfo.php:
```diff
@@ -155,6 +155,19 @@ public function &getInfo()
+	public function phpinfo_enabled()
+	{
+		$disabled = explode(',', ini_get('disable_functions'));
+		return !in_array('phpinfo', $disabled);
```
@pcmitsis can you add one clear line bevor the return?

pcmitsis committed:
Fixes from  @zero-24 comments on PR #6148

pcmitsis commented:
@zero-24  I fixed them both. I wasn't sure about the version that's why i had 3.5 ! Sorry for the inconvenience.

zero-24 commented:
@pcmitsis no Problem :smile:

pcmitsis committed:
Fix second whitespace

pcmitsis commented:
@zero-24  Now it should be fine!

zero-24 committed:
Update sysinfo.php

zero-24 committed:
Update en-GB.com_admin.ini

pcmitsis committed:
Merge pull request #2 from zero-24/patch-8

Add a message if phpinfo is disabled.

zero-24 commented on administrator/components/com_admin/models/sysinfo.php:
```diff
@@ -155,6 +155,20 @@ public function &getInfo()
+	{
+		$disabled = explode(',', ini_get('disable_functions'));
+
+		return in_array('phpinfo', $disabled);
```
@pcmitsis can you please re add the `!` here. I remove this only for testing and don't re add them on my pull sorry.

zero-24 commented:
@brianteeman can you have a look into the new string?

pcmitsis commented:
@zero-24 Thank you for you enhancement. Your logic about the issue was better than mine. As a newbie i didn't knew where should i add the error message. Thank you so much :smiley:

pcmitsis committed:
Add a message if phpinfo is disabled. Fix for @zero-24.

zero-24 commented:
@test success full with last commit. Thanks @pcmitsis <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6148">issues.joomla.org/joomla-cms/6148</a>.</sub>

brianteeman commented on administrator/language/en-GB/en-GB.com_admin.ini:
```diff
@@ -114,6 +114,7 @@ COM_ADMIN_PHP_BUILT_ON="PHP Built On"
 COM_ADMIN_PHP_INFORMATION="PHP Information"
 COM_ADMIN_PHP_SETTINGS="PHP Settings"
 COM_ADMIN_PHP_VERSION="PHP Version"
+COM_ADMIN_PHPINFO_DISABLED="The build in phpinfo() function is disabled by your host."
```
The correct english for that string would be 
The built in phpinfo() function has been disabled by your host.

However is this really the correct thing to say. The host has not disabled the function in joomla. They have disabled the  fucntion in php (on the server) so Joomla can not display the information

brianteeman commented:
I have commented in line about the string. Both correcting the english of the string used and questioning if it is even the correct thing to say

pcmitsis commented:
@brianteeman should i change the string with a new commit with the one proposed?

zero-24 commented:
>should i change the string with a new commit with the one proposed?

@pcmitsis yes. Brian is a en-GB native speaker and the lead of the en-GB language file WG.

Thanks @brianteeman

pcmitsis committed:
Fix error message when phpinfo() is called on server that disables this function

This is the final commit for PR #6148. I would like to thank @zero-24 and @brianteeman for their valuable
help!

pcmitsis commented:
Thank you @brianteeman and @zero-24  for your valuable help in this PR!

larsomator commented:
@pcmitsis Hi, thanks for your work with my bug report.

pcmitsis commented:
@larsomator you are welcome!

zero-24 commented:
@larsomator can you test @pcmitsis 's PR? So we can move this to RTC. (and maybe merge for 3.4.1 :smile: )

if you need help for testing the com_patchtester will help you:
https://docs.joomla.org/Testing_Joomla!_patches
https://docs.joomla.org/Component_Patchtester_for_Testers

Thanks!

Kubik-Rubik commented on administrator/components/com_admin/models/sysinfo.php:
```diff
@@ -327,8 +345,7 @@ public function &getEditor()
 		{
-			$config = JFactory::getConfig();
-			$this->editor = $config->get('editor');
+			$this->editor = JFactory::getConfig()->get('editor');
```
Please do such code optimizations in a separated PR next time. Change only related code parts of the specific PR!

Kubik-Rubik commented on administrator/components/com_admin/models/sysinfo.php:
```diff
@@ -163,7 +177,7 @@ public function &getInfo()
 	public function &getPHPInfo()
 	{
-		if (is_null($this->php_info))
+		if (is_null($this->php_info) and $this->phpinfo_enabled())
```
Please change "and" to "&&".

pcmitsis committed:
Update sysinfo.php

@Kubik-Rubik change from and to &&

pcmitsis commented:
@Kubik-Rubik thank you very much for your notes! I already changed the `and` to `&&`.Would you like me to do anything else?

Kubik-Rubik commented:
@pcmitsis Thanks! Could you please change the function name from "phpinfo_enabled" to "phpinfoEnabled" (and of course the call of this function). Then we can merge you PR! :-)

pcmitsis committed:
Fix for issue #6042

pcmitsis commented:
@Kubik-Rubik done! Travis CI build is in progress :smile:

Kubik-Rubik commented:
Thank you @pcmitsis!
